OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -249,80 +249,83 @@ public void ResolveOrder(Actor self, Order order)
+							return;
+					}
 
-						var amountToBuild = Math.Min(fromLimit, order.ExtraData);
-						for (var n = 0; n < amountToBuild; n++)
+					var valued = unit.TraitInfoOrDefault<ValuedInfo>();
+					var cost = valued != null ? valued.Cost : 0;
+					var time = GetBuildTime(unit);
+					var amountToBuild = Math.Min(fromLimit, order.ExtraData);
+					for (var n = 0; n < amountToBuild; n++)
+					{
+						var hasPlayedSound = false;
+						BeginProduction(new ProductionItem(this, order.TargetString, cost, playerPower, () => self.World.AddFrameEndTask(_ =>
 						{
-							var hasPlayedSound = false;
-							BeginProduction(new ProductionItem(this, order.TargetString, cost, playerPower, () => self.World.AddFrameEndTask(_ =>
+							var isBuilding = unit.HasTraitInfo<BuildingInfo>();
+
+							if (isBuilding && !hasPlayedSound)
+								hasPlayedSound = Game.Sound.PlayNotification(rules, self.Owner, "Speech", Info.ReadyAudio, self.Owner.Faction.InternalName);
+							else if (!isBuilding)
 							{
-								var isBuilding = unit.HasTraitInfo<BuildingInfo>();
-
-								if (isBuilding && !hasPlayedSound)
-									hasPlayedSound = Game.Sound.PlayNotification(rules, self.Owner, "Speech", Info.ReadyAudio, self.Owner.Faction.InternalName);
-								else if (!isBuilding)
-								{
-									if (BuildUnit(order.TargetString))
-										Game.Sound.PlayNotification(rules, self.Owner, "Speech", Info.ReadyAudio, self.Owner.Faction.InternalName);
-									else if (!hasPlayedSound && time > 0)
-										hasPlayedSound = Game.Sound.PlayNotification(rules, self.Owner, "Speech", Info.BlockedAudio, self.Owner.Faction.InternalName);
-								}
-							})));
-						}
-
-						break;
+								if (BuildUnit(order.TargetString))
+									Game.Sound.PlayNotification(rules, self.Owner, "Speech", Info.ReadyAudio, self.Owner.Faction.InternalName);
+								else if (!hasPlayedSound && time > 0)
+									hasPlayedSound = Game.Sound.PlayNotification(rules, self.Owner, "Speech", Info.BlockedAudio, self.Owner.Faction.InternalName);
+							}
+						})));
 					}
 
+					break;
 				case "PauseProduction":
-					{
-						if (queue.Count > 0 && queue[0].Item == order.TargetString)
-							queue[0].Pause(order.ExtraData != 0);
-
-						break;
-					}
+					if (queue.Count > 0 && queue[0].Item == order.TargetString)
+						queue[0].Pause(order.ExtraData != 0);
 
+					break;
 				case "CancelProduction":
-					{
-						CancelProduction(order.TargetString, order.ExtraData);
-						break;
-					}
+					CancelProduction(order.TargetString, order.ExtraData);
+					break;
 			}
 		}
 
 		public virtual int GetBuildTime(string unitString)
 		{
+			if (self.World.AllowDevCommands && self.Owner.PlayerActor.Trait<DeveloperMode>().FastBuild)
+				return 0;
+
 			var unit = self.World.Map.Rules.Actors[unitString];
-			if (unit == null || !unit.HasTraitInfo<BuildableInfo>())
+			if (unit == null)
 				return 0;
 
+			var time = unit.GetBuildTime() * Info.BuildSpeed;
+			return (int)time;
```

Comment:
You could just `return GetBuildTime(unit)` here. Move the null check up and remove the cheat check.